src/componants/Articles.jsx

```diff
@@ -1,53 +1,82 @@
 import { useContext, useEffect, useState } from "react";
 import { getArticleById, getArticles, getRandomArticleId } from "../../api";
 
-import {ArticleCard, RandomArticleCard}from "./ArticleCard";
+import { ArticleCard, RandomArticleCard } from "./ArticleCard";
 import { useParams } from "react-router-dom";
 import { useCol } from "react-bootstrap/esm/Col";
 import categoryContext from "../contexts/categoryContexts";
+import { useSearchParams } from "react-router-dom";
+import * as React from "react";
+
+import Dropdown from "react-bootstrap/Dropdown";
+import DropdownButton from "react-bootstrap/DropdownButton";
+
 
 function Articles() {
 	const [articles, setArticles] = useState([]);
-	const [randomArticle, setRandomArticle] = useState({});
-  const {topic} = useParams()
-  const [loaded, setLoaded] = useState(false)
-  const {setCategory} = useContext(categoryContext)
+	const { topic } = useParams();
+	const [loaded, setLoaded] = useState(false);
+	const { setCategory } = useContext(categoryContext);
+  const [sortBy, setSortBy] = useState("");
+  const [order, setOrder] = useState("")
+  const [searchParams, setSearchParams] = useSearchParams();
 
 	useEffect(() => {
-    if(topic){
-      setCategory(topic)
+    const searchObject = {}
+		if (topic) {
+			setCategory(topic);
+		}
+    if(sortBy !== ''){
+      searchObject.sort_by = sortBy
     }
-		getArticles(topic)
-			.then((results) => {
-				setArticles(results);
-        setLoaded(true)
-			})
-			// .then((results) => {
-			// return	getRandomArticleId(results.length);
-			// })
-			// .then((result) => {
-      //   console.log(result)
-      //   console.log(articles)
-			// return	getArticleById(articles[result].article_id);
-			// })
-			// .then((result) => {
-			// 	setRandomArticle(result);
-      //   console.log(randomArticle)
-			// });
-	}, [topic]);
-
-if(loaded){
-	return (
-		<>
-			<section className="resultsFlex">
-				{articles.map((article) => {
-					return <ArticleCard article={article} key={article.article_id} />;
-				})}
-			</section>
-		</>
-	);
-}
-else return <p>Loading...</p>
+    if(order !== ''){
+      searchObject.order = order
+    }
+		getArticles(topic, sortBy === '' ? undefined : sortBy, order === '' ? undefined : order).then((results) => {
```

Comment:
There's no need for default values. The Axis params object will only add them to the query is they are not undefined